### PR TITLE
Supressed reload dialog when Monitoring is enabled  #3142

### PR DESCRIPTION
**Pull request by jrocha**
When Monitoring is enabled N++ was asking for a file reload.

### CODE REVIEW

**Comment by donho** [comment]
why do we need 4ce140d ?

**Comment by donho** [comment]
Is there a simple way to reproduce the bug?

**Comment by donho** [comment]
@jrocha This PR is closed due to the absence of your reply.
Feel free to open it with your answer.

**Comment by dougbreaux** [comment]
@donho I just found this while looking to see if I should open an Issue specifically just to "suppress reload dialog when Monitoring is enabled". If this particular PR isn't going anywhere, should I open a separate issue just for that? I haven't found one that matches just that case.

(Maybe commit https://github.com/notepad-plus-plus/notepad-plus-plus/pull/4527/commits/ef26ff66d65e49c1ba121b38a922f0f9e104dc06 ?)

(My case is Java log4j-written files. Sometimes when the files are appended the warning occurs, sometimes it doesn't, but it's very frequent & reproduceable that it does. Maybe when the Java process restarts entirely, or maybe tied to some difference in certain log4j internal paths.)

**Comment by jrocha** [comment]
Hi @donho, sorry about my delay. I've explained the need of an active checker over [here](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/3142).

We need [4ce140d](https://github.com/notepad-plus-plus/notepad-plus-plus/commit/4ce140ddfcbd5efa5a4741b47150f7bc0c65381b) because Windows isn't honoring FILE_NOTIFY_CHANGE_SIZE or FILE_NOTIFY_CHANGE_LAST_WRITE correctly if the file writer keep it opened.

To reproduce the bug I've been using a simple [C# code](https://gist.github.com/jrocha/1fd533ae51bbe9e91278747e1b304dc6). You just have to write to a file, flush and keep it open.

**Comment by ndrewl** [comment]
@jrocha, could you please reopen this pull request, as @donho suggested: 

> Feel free to open it with your answer.

Then maybe it will have a chance to get into release.

**Comment by jrocha** [comment]
@ndrewl I'm afraid I don't how to do that. Just googled and it seems to not be possible [link](https://stackoverflow.com/questions/12674304/github-reopening-a-merged-pull-request). @donho  should I create a PR?

**Comment by Rudolpher** [comment]
Like others, just learned of this feature then realize it doesn't update without focus. Is there any movement on this, or better alternatives than UnxUtils with tail?

**Comment by chcg** [comment]
@jrocha I just reopened it for you. There should have been a button with "reopen pull request" down here.

File: PowerEditor/src/WinControls/ReadDirectoryChanges/ReadFileChanges.h [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+#include <windows.h>
+
+
+class CReadFileChanges
+{
+public:
+	CReadFileChanges();
+	~CReadFileChanges();
+	void AddFile(LPCTSTR szDirectory, DWORD dwNotifyFilter);
+	BOOL CReadFileChanges::DetectChanges();
```
Comment (SinghRajenM) [inline PowerEditor/src/WinControls/ReadDirectoryChanges/ReadFileChanges.h]: class name is not required here.

File: PowerEditor/src/WinControls/ReadDirectoryChanges/ReadFileChanges.h [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+	void Terminate();
+
+private:
+	LPCTSTR szFile;
```
Comment (SinghRajenM) [inline PowerEditor/src/WinControls/ReadDirectoryChanges/ReadFileChanges.h]: Member variables should be initialized either in class member initializer or in contructor.

File: PowerEditor/src/WinControls/ReadDirectoryChanges/ReadFileChanges.h (hunk 1 above)
Comment (jrocha) [reply PowerEditor/src/WinControls/ReadDirectoryChanges/ReadFileChanges.h]: Solved. Thanks.

File: PowerEditor/src/WinControls/ReadDirectoryChanges/ReadFileChanges.h (hunk 2 above)
Comment (jrocha) [reply PowerEditor/src/WinControls/ReadDirectoryChanges/ReadFileChanges.h]: Solved. Thanks.

**Comment by jrocha** [comment]
@chcg Thanks for reopening this PR. I've tried to switch branches but looks like since this PR has started on master it cannot be changed without a new PR. I'm going to use a separated branch next time.

There are a few extra commits over here because I've mistakenly commited a wrong file and had a hard time reverting it.

**Comment by jrocha** [comment]
@donho @chcg  any news on this PR?

**Comment by donho** [comment]
@jrocha 
Please close this issue and create a new one since this one has the conflict with the current codes.
Please alse provide an easy way to reproduce it to valid your PR.

**Comment by donho** [comment]
This PR is closed due to the absence of authors reply. 
Please make a new PR, then I will check it ASAP.

**Pull request closed without merging**